vxyowsup/whatsapp.py

```diff
@@ -277,3 +283,8 @@ def onEvent(self, event):
         elif name == YowNetworkLayer.EVENT_STATE_DISCONNECTED:
             reactor.callFromThread(
                 self.transport.handle_disconnected, event.args.get('reason'))
+            self.broadcastEvent(
+                YowLayerEvent(YowNetworkLayer.EVENT_STATE_CONNECT))
```

Comment:
This way we send an event down to the network layer, which handles connecting again. I'm unsure how to properly test this, but leaving the transport running for a while shows it disconnecting and reconnecting again. I can't properly test this, besides the unit tests, as yowsup is currently very broken.